File: Problems/FixTheCode/FTC_3_sorting-algorithms/python/source/selection_sorter.py
from customSorter import CustomSorter
from arrays import swapTwoNumbers

class SelectionSorter(CustomSorter):

    def sort(self):
        print("Sorting array using SELECTION SORT.")
        array_len = len(self.array)

        for i in range(array_len):
            min_index = i
            for j in range(i + 1, array_len):
                if self.array[j] > self.array[min_index]:
                    min_index = j
            swapTwoNumbers(self.array, i, min_index)

File: Problems/FixTheCode/FTC_3_sorting-algorithms/java/source/SelectionSorter.java
public class SelectionSorter implements Sorter {

    public SelectionSorter(){}

    public int[] Sort(int[] array){
        System.out.println("Sorting array using SELECTION SORT.");

        int[] sortedArray = array.clone();

        for (int i=0; i<sortedArray.length-1; i++)
        {
            int minIndex = i;
            for (int j=i+1; j<sortedArray.length-2; j++){
                if (sortedArray[j] > sortedArray[minIndex]){
                    minIndex = j;
                }
            }
            Arrays.SwapTwoNumbers(sortedArray, i, minIndex);
        }
        return sortedArray;
    }
}
